tests/parity/toolcalling/capture_toolcalling_outputs.py

```
@@ -414,6 +439,126 @@ def merge_into_fixtures(
+    container: str, impl: str, requests: list[dict]
+) -> dict[str, dict]:
+    """Copy the worker bundle into `container` and run every request there once.
+
+    Returns {key: {calls, normal_text, error}}. The engine import happens a
+    single time inside the worker, so the whole batch shares one heavy startup.
+    Works for devcontainers and bare engine images alike (nothing is assumed
+    about a mounted repo — the bundle is copied in).
+    """
+    dest = "/tmp/parity_worker"
+    files = _BUNDLE + [f"tests/parity/toolcalling/{impl}.py"]
+    subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)